ui/app/AppLayouts/Wallet/popups/swap/SwapModal.qml

```diff
@@ -32,10 +33,31 @@ StatusDialog {
     rightPadding: Style.current.xlPadding
     backgroundColor: Theme.palette.baseColor3
 
+    QtObject {
+        id: d
+         property var fetchSuggestedRoutes: Backpressure.debounce(root, 1000, function() {
+                 root.swapAdaptor.fetchSuggestedRoutes()
+         })
+    }
+
+    Connections {
+        target: root.swapInputParamsForm
+        function onFormValuesChanged() {
+            if(root.swapInputParamsForm.isFormFilledCorrectly()) {
+                root.swapAdaptor.swapProposalLoading = true
+                d.fetchSuggestedRoutes()
+            } else {
+                root.swapAdaptor.swapProposalLoading = false
```

Comment:
Hmm if we don't cancelling the `debounce` timer when we reach this point, the backend will be called with bad data right after the timer expires righr? I don't think there's a convenient way to cancel the timer, so perhaps the inputParams validation should move to the SwapAdaptor instead?

Reply:
good point! moved now @dlipicar